packages/loot-core/src/server/budget/goaltemplates.js

```diff
@@ -333,6 +350,42 @@ async function applyCategoryTemplate(category, template_lines, month, force) {
       case 'error':
         return null;
       default:
+      case 'schedule': {
+        let { id: schedule_id } = await db.first(
+          'SELECT id FROM schedules WHERE name = ?',
+          [template.name],
+        );
+        let rule = await getRuleForSchedule(schedule_id);
+        let conditions = rule.serialize().conditions;
+        let { date: dateCond, amount: amountCond } =
+          extractScheduleConds(conditions);
+        let next_date_string = getNextDate(dateCond, current_month);
+        let num_months = differenceInCalendarMonths(
+          new Date(next_date_string),
+          current_month,
+        );
+        let target = -getScheduledAmount(amountCond.value);
+        let diff = target - last_month_balance;
+        if (num_months > 0) {
+          if (diff >= 0 && num_months > -1) {
+            to_budget += Math.round(diff / (num_months + 1));
```

Comment:
This is where the the monthly allocation error seems to be that I'm seeing.
```suggestion
            to_budget += Math.round(diff / num_months);
```

Reply:
Yes, this should probably fix it. Brainless copy-paste gone wrong, although I still don't quite get why I could not replicate that error